src/Spice86.Shared/Emulator/VM/Breakpoint/Serializable/SerializedBreakpoint.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace Spice86.Shared.Emulator.VM.Breakpoint.Serializable;
+
+using System.Text.Json.Serialization;
+
+/// <summary>
+/// Base class for serializable breakpoint data.
+/// </summary>
+[JsonPolymorphic(TypeDiscriminatorPropertyName = "$type")]
+[JsonDerivedType(typeof(SerializedBreakpoint), typeDiscriminator: "breakpoint")]
+[JsonDerivedType(typeof(SerializedBreakpointRange), typeDiscriminator: "range")]
+public record SerializedBreakpoint {
+    /// <summary>
+    /// Gets the format version.
+    /// </summary>
+    public int Version { get; init; } = 1;
+
+    /// <summary>
+    /// Gets the trigger value for the breakpoint.
+    /// </summary>
+    public long Trigger { get; init; }
+
+    /// <summary>
+    /// Gets the type of the breakpoint.
+    /// </summary>
+    public BreakPointType Type { get; init; }
+
+    /// <summary>
+    /// Gets a value indicating whether the breakpoint should be removed after being triggered.
+    /// </summary>
+    public bool IsRemovedOnTrigger { get; init; }
+
+    /// <summary>
+    /// Gets the comment for the breakpoint.
+    /// </summary>
+    public string Comment { get; init; } = string.Empty;
```

Comment:
This doesnt seem to be user fillable. Only use case I could find was the step over breakpoint.

By the way, do we serialize step over breakpoints with this code? I am not sure we would because they are supposed to be ephemeral usually, but what if step over is never reached and we save? Do we get them in the dump?